### PR TITLE
[fs-extra] Update to v11

### PR DESCRIPTION
**Pull request by BendingBender**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/jprichardson/node-fs-extra/blob/master/CHANGELOG.md
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/fs-extra/test/fs-extra-tests-module.ts [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+type CopyFilterSync = fs.CopyFilterSync;
+type CopyFilterAsync = fs.CopyFilterAsync;
+type SymlinkType = fs.SymlinkType;
+type CopyOptions = fs.CopyOptions;
+type CopyOptionsSync = fs.CopyOptionsSync;
+type EnsureOptions = fs.EnsureOptions;
+type MoveOptions = fs.MoveOptions;
+type ReadOptions = fs.ReadOptions;
+type WriteFileOptions = fs.WriteFileOptions;
+type WriteOptions = fs.WriteOptions;
```
Comment (peterblazejewicz) [inline types/fs-extra/test/fs-extra-tests-module.ts:13]: Can we test the same (test types) with an import expected to be used in ESM?
```suggestion
import fs, {
    CopyFilterAsync,
    CopyFilterSync,
    CopyOptions,
    CopyOptionsSync,
    EnsureOptions,
    MoveOptions,
    ReadOptions,
    SymlinkType,
    WriteFileOptions,
    WriteOptions,
} from 'fs-extra/esm.mjs';

// test type exports
type Types =
    | CopyFilterSync
    | CopyFilterAsync
    | CopyOptions
    | CopyOptionsSync
    | EnsureOptions
    | MoveOptions
    | ReadOptions
    | SymlinkType
    | WriteFileOptions
    | WriteOptions;
```

File: types/fs-extra/OTHER_FILES.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+esm.d.mts
```
Comment (peterblazejewicz) [inline types/fs-extra/OTHER_FILES.txt:1]: I'm not sure if that is required, if proper ESM import would be used in module test

Comment (peterblazejewicz) [reply types/fs-extra/OTHER_FILES.txt:1]: pls disregard, still required when testing TS configs

File: types/fs-extra/test/fs-extra-tests-module.ts (hunk 1 above)
Comment (BendingBender) [reply types/fs-extra/test/fs-extra-tests-module.ts]: done

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@BendingBender thanks!

**Comment by BendingBender** [comment]
Ready to merge

**Comment by zheung** [comment]
vscode does not seem to recognize properly when using the `esm` module
image

If you enter `/esm.mjs`, it cannot be imported and run correctly, 
image
but it can correctly show
image

Refer to package.json, `esm` is set in the `export` option. It is not a real file.
image

**Comment by BendingBender** [comment]
Working on it, but still haven't found a satisfactory solution: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63859

**Comment by BendingBender** [comment]
@zheung What value do you use for the `module` compiler option in your `tsconfig.json`? It must be `"module": "Node16"`, otherwise it won't work. And you must use `import { readJSONSync } from 'fs-extra/esm'` as your import statement (Note: it _mustn't_ have the `.mjs` suffix).

Also note that there's still a bug in the definitions, it'll be fixed when https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63859 is merged. In the current version, all exports from `fs-extra/esm` will be `any`-typed.

**Comment by zheung** [comment]
> @zheung What value do you use for the `module` compiler option in your `tsconfig.json`? It must be `"module": "Node16"`, otherwise it won't work. And you must use `import { readJSONSync } from 'fs-extra/esm'` as your import statement (Note: it _mustn't_ have the `.mjs` suffix).
> 
> Also note that there's still a bug in the definitions, it'll be fixed when #63859 is merged. In the current version, all exports from `fs-extra/esm` will be `any`-typed.

`ESNext`. I will test whether my compiler options affect the tips in the next version.
````json
{
	"compilerOptions": {
		"module": "ESNext",
		"moduleResolution": "Node",
		"target": "ESNext",
		"experimentalDecorators": true,
		"strict": true
	},
	"exclude": [
		"node_modules",
		"dist"
	]
}
````


When "/" is entered, the `esm.mjs` option will be displayed. I know this is incorrect, just for your reference :)
image

**Comment by BendingBender** [comment]
It won't as soon as you switch to `"module": "Node16"`.

**Comment by BendingBender** [comment]
`moduleResolution` must also be set to `Node16`. But strictly seen, you don't need to set it, the correct setting will be picked when you remove it.

**Comment by peterblazejewicz** [comment]
PR #63859 got merged

**Pull request merged**